### PR TITLE
Update to React 0.14

### PR DESCRIPTION
**Pull request by rolyatmax**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,9 +23,9 @@ Once you have the repository cloned run the following commands to get started:
 
 ```shell
 npm install
-gulp develop
+npm run develop
```
Comment (AlanFoster) [inline README.md]: :+1: I like this opinionated change :)

Comment (rolyatmax) [inline README.md]: Thanks! :-)

**Comment by AlanFoster** [comment]
@rolyatmax Thanks for the work; Could we split the first two commits (cdf0ef09e5778e356b170c0e721e408a817d4389 and 6867bd59407c5790792e85e85c54901d1ed5d6c0) into different PRs so they can be speed-merged.

For visibility - there's a similar PR to upgrade to 0.14 here #77. We should combine the efforts of these PRs if possible

File: package.json [hunk 2]
```diff
@@ -24,28 +28,34 @@
     "url": "https://github.com/onefinestay/react-daterange-picker"
   },
   "peerDependencies": {
-    "react": ">=0.12.0"
+    "react": ">=0.14.0"
```
Comment (AlanFoster) [inline package.json]: We should add `react-dom` as a peer dependency too

File: package.json [hunk 3]
```diff
@@ -24,28 +28,34 @@
...
   },
   "dependencies": {
     "calendar": "^0.1.0",
     "classnames": "^2.1.1",
     "immutable": "^3.7.2",
     "moment": "^2.10.6",
-    "moment-range": "^2.0.3"
+    "moment-range": "^2.0.3",
+    "react": "^0.14.0",
```
Comment (AlanFoster) [inline package.json]: We should remove `react` and `react-dom` from our dependencies, if they will be provided as peerDependencies

File: package.json [hunk 4]
```diff
@@ -24,28 +28,34 @@
...
+    "react-addons-pure-render-mixin": "^0.14.0",
```
Comment (AlanFoster) [inline package.json]: :+1: For keeping this as a dependency

**Comment by rolyatmax** [comment]
Closing in favor of #79 and #80

**Pull request closed without merging**